File: backend/app/models/augment.py
from sqlalchemy import Enum, String, Text
from sqlalchemy.dialects.postgresql import JSONB
from sqlalchemy.orm import Mapped, mapped_column

from app.database import Base


class Augment(Base):
    __tablename__ = "augments"

    id: Mapped[str] = mapped_column(String, primary_key=True)
    name: Mapped[str] = mapped_column(String, nullable=False)
    tier: Mapped[str] = mapped_column(
        Enum("Silver", "Gold", "Prismatic", name="augment_tier_enum"), nullable=False
    )
    description: Mapped[str | None] = mapped_column(Text, nullable=True)
    image_url: Mapped[str | None] = mapped_column(String, nullable=True)
    comp_synergy: Mapped[list] = mapped_column(JSONB, nullable=False, default=list)

File: flutter_app/lib/models/augment.dart
// Augment 모델
// 스키마 기준: contracts/api-spec.yaml

class Augment {
  final String id;
  final String name;
  final String tier; // Silver, Gold, Prismatic
  final String? description;
  final String? imageUrl;
  final List<String> compSynergy;

  const Augment({
    required this.id,
    required this.name,
    required this.tier,
    this.description,
    this.imageUrl,
    required this.compSynergy,
  });

  factory Augment.fromJson(Map<String, dynamic> json) {
    return Augment(
      id: json['id'] as String,
      name: json['name'] as String,
      tier: json['tier'] as String,
      description: json['description'] as String?,
      imageUrl: json['image_url'] as String?,
      compSynergy: (json['comp_synergy'] as List<dynamic>?)
              ?.map((e) => e.toString())
              .toList() ??
          [],
    );
  }

  Map<String, dynamic> toJson() => {
        'id': id,
        'name': name,
        'tier': tier,
        'description': description,
        'image_url': imageUrl,
        'comp_synergy': compSynergy,
      };
}
